src/themes/JSONTheme.ts

```diff
@@ -0,0 +1,1003 @@
+    protected blackShadowImages: CanvasImageSource[] = [];
+    protected matrices: Array<MatrixStore> = [];
+
+    constructor(parent?: GoTheme) {
+        super();
+        this.name = "JSONTheme"; //JSONTheme.styles.themeName
+        this.parent = parent;
+
+        if (!JSONTheme.styles) {
+            JSONTheme.styles = new JSONThemeStyle();
+        }
+
+        if (JSONTheme.json && JSONTheme.json.length > 4) {
+            this.loadFromText(JSONTheme.json);
+        } else {
+            this.loadFromText(JSONTheme.getDefaultJSON());
+        }
+    }
+
+    public rebuildImages() {
+        this.whiteImages = [];
+        this.blackImages = [];
+        this.whiteShadowImages = [];
+        this.blackShadowImages = [];
+
+        if (JSONTheme.styles.whiteStones && JSONTheme.styles.whiteStones.length > 0) {
+            for (const src of JSONTheme.styles.whiteStones) {
+                const img = new Image();
+                img.src = src;
+                this.whiteImages.push(img);
+            }
+        }
+
+        if (JSONTheme.styles.blackStones && JSONTheme.styles.blackStones.length > 0) {
+            for (const src of JSONTheme.styles.blackStones) {
+                const img = new Image();
+                img.src = src;
+                this.blackImages.push(img);
+            }
+        }
+
+        if (JSONTheme.styles.whiteShadows && JSONTheme.styles.whiteShadows.length > 0) {
+            for (const src of JSONTheme.styles.whiteShadows) {
+                const img = new Image();
+                img.src = src;
+                this.whiteShadowImages.push(img);
+            }
+        }
+
+        if (JSONTheme.styles.blackShadows && JSONTheme.styles.blackShadows.length > 0) {
+            for (const src of JSONTheme.styles.blackShadows) {
+                const img = new Image();
+                img.src = src;
+                this.blackShadowImages.push(img);
+            }
+        }
+
+        // created default shadows if needed
+        if (JSONTheme.styles.shadows && JSONTheme.styles.shadows.length > 0) {
+            if (this.whiteShadowImages.length === 0) {
+                for (const src of JSONTheme.styles.shadows) {
+                    const img = new Image();
+                    img.src = src;
+                    this.whiteShadowImages.push(img);
+                }
+            }
+            if (this.blackShadowImages.length === 0) {
+                for (const src of JSONTheme.styles.shadows) {
+                    const img = new Image();
+                    img.src = src;
+                    this.blackShadowImages.push(img);
+                }
+            }
+        }
+    }
+
+    public rebuildMatrices(seed: number) {
+        this.matrices = JSONTheme.buildMatrices(seed);
+    }
+    protected static buildMatrices(seed: number): Array<MatrixStore> {
+        // have to use static method because theme changes arent honored due to pre-rendered stone caching
```

Comment:
Sounds like the cache needs to be purged?

Reply:
Yeah I think that comment was because I was trying to buildMatrices() in "prerenderStone" which wasn't working. Now I build them in the constructor, so the static method isn't necessary.

Fixed with [b362b53](https://github.com/online-go/goban/pull/63/commits/b362b53bcc7bddcb5ec28fde0c4d0fcb9857b1e8)